Investigate pytest issues on CI

CI run logs: [logs_5403.zip](https://github.com/hikari-py/hikari/files/8015710/logs_5403.zip)


Failing pipeline: `test (macos-latest, 3.9)`

2 errors I found:

1. Warning raised by slow tests, fixed by #1005, as it rewrites the test. These should have been test errors but they were captured as thread warnings instead unfortunately, so it took some time to catch. @FasterSpeeding this is the test you told me about. Ig I fixed it by mistake 😅 
```
2022-02-07T13:17:06.0187500Z tests/hikari/impl/test_rate_limits.py::TestWindowedBurstRateLimiter::test_throttle_when_limited_sleeps_then_bursts_repeatedly
2022-02-07T13:17:06.0188250Z   /Users/runner/work/hikari/hikari/.nox/pytest/lib/python3.9/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-5
2022-02-07T13:17:06.0188630Z   
2022-02-07T13:17:06.0188810Z   Traceback (most recent call last):
2022-02-07T13:17:06.0202960Z     File "/Users/runner/hostedtoolcache/Python/3.9.10/x64/lib/python3.9/threading.py", line 973, in _bootstrap_inner
2022-02-07T13:17:06.0203420Z       self.run()
2022-02-07T13:17:06.0203790Z     File "/Users/runner/hostedtoolcache/Python/3.9.10/x64/lib/python3.9/threading.py", line 910, in run
2022-02-07T13:17:06.0204110Z       self._target(*self._args, **self._kwargs)
2022-02-07T13:17:06.0205110Z     File "/Users/runner/work/hikari/hikari/tests/hikari/impl/test_rate_limits.py", line 343, in _run_test_throttle_logic_on_this_thread
2022-02-07T13:17:06.0205510Z       event_loop.run_until_complete(self._run_test_throttle_logic_on_loop(event_loop))
2022-02-07T13:17:06.0205950Z     File "/Users/runner/hostedtoolcache/Python/3.9.10/x64/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
2022-02-07T13:17:06.0206270Z       return future.result()
2022-02-07T13:17:06.0206650Z     File "/Users/runner/work/hikari/hikari/tests/hikari/impl/test_rate_limits.py", line 396, in _run_test_throttle_logic_on_loop
2022-02-07T13:17:06.0207340Z       assert math.isclose(next_first - previous_last, period, abs_tol=max_distance_within_window), (
2022-02-07T13:17:06.0207720Z   AssertionError: distance between windows is not acceptable! 0=39.98175229 1=41.582799405, max diff = 0.05
2022-02-07T13:17:06.0207980Z   assert False
2022-02-07T13:17:06.0208390Z    +  where False = <built-in function isclose>((41.582799405 - 39.98175229), 1.5, abs_tol=0.05)
2022-02-07T13:17:06.0208780Z    +    where <built-in function isclose> = math.isclose
2022-02-07T13:17:06.0209290Z   
2022-02-07T13:17:06.0209580Z     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
2022-02-07T13:17:06.0209800Z 
2022-02-07T13:17:06.0209970Z tests/hikari/test_channels.py::TestTextChannel::test_trigger_typing
2022-02-07T13:17:06.0210810Z   tests/hikari/test_channels.py:302: PytestWarning: The test <Function test_trigger_typing> is marked with '@pytest.mark.asyncio' but it is not an async function. Please remove asyncio marker. If the test is not marked explicitly, check for global markers applied via 'pytestmark'.
2022-02-07T13:17:06.0211330Z     def test_trigger_typing(self, model):
2022-02-07T13:17:06.0211480Z 
2022-02-07T13:17:06.0211760Z -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

2. Apart from the repr for `TimedCacheMap` being horrible for debugging this, it looks like another speed issue. #1005 also fixes this by completely removing the tests for it. If we were to keep it, I think there is also a way to not make it flaky
```
2022-02-07T13:17:05.9862020Z ____________ TestTimedCacheMap.test___setitem___garbage_collection _____________
2022-02-07T13:17:05.9862570Z 
2022-02-07T13:17:05.9863100Z self = <tests.hikari.internal.test_collections.TestTimedCacheMap object at 0x1066f03a0>
2022-02-07T13:17:05.9863970Z 
2022-02-07T13:17:05.9864460Z     @pytest.mark.asyncio()
2022-02-07T13:17:05.9864990Z     async def test___setitem___garbage_collection(self):
2022-02-07T13:17:05.9865520Z         mock_map = collections.TimedCacheMap(
2022-02-07T13:17:05.9866120Z             expiry=datetime.timedelta(seconds=hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 3)
2022-02-07T13:17:05.9866660Z         )
2022-02-07T13:17:05.9867100Z         mock_map["OK"] = "no"
2022-02-07T13:17:05.9868290Z         await asyncio.sleep(hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 2)
2022-02-07T13:17:05.9868850Z         assert mock_map == {"OK": "no"}
2022-02-07T13:17:05.9869350Z         mock_map["ayanami"] = "rei"
2022-02-07T13:17:05.9869850Z >       assert mock_map == {"OK": "no", "ayanami": "rei"}
2022-02-07T13:17:05.9870780Z E       AssertionError: assert <hikari.inter...t 0x1060cff40> == {'OK': 'no', 'ayanami': 'rei'}
2022-02-07T13:17:05.9871360Z E         Full diff:
2022-02-07T13:17:05.9871960Z E         - {'OK': 'no', 'ayanami': 'rei'}
2022-02-07T13:17:05.9872580Z E         + <hikari.internal.collections.TimedCacheMap object at 0x1060cff40>
2022-02-07T13:17:05.9873010Z 
2022-02-07T13:17:05.9873200Z mock_map   = <hikari.internal.collections.TimedCacheMap object at 0x1060cff40>
2022-02-07T13:17:05.9873600Z self       = <tests.hikari.internal.test_collections.TestTimedCacheMap object at 0x1066f03a0>
2022-02-07T13:17:05.9873820Z 
2022-02-07T13:17:05.9873960Z tests/hikari/internal/test_collections.py:251: AssertionError
2022-02-07T13:17:05.9874490Z ------------------------------ Captured log call -------------------------------
2022-02-07T13:17:05.9876750Z WARNING  asyncio:base_events.py:1885 Executing <Task pending name='Task-5' coro=<TestTimedCacheMap.test___setitem___garbage_collection() running at /Users/runner/work/hikari/hikari/tests/hikari/internal/test_collections.py:248> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x105b67c10>()] created at /Users/runner/hostedtoolcache/Python/3.9.10/x64/lib/python3.9/asyncio/base_events.py:424> cb=[_run_until_complete_cb() at /Users/runner/hostedtoolcache/Python/3.9.10/x64/lib/python3.9/asyncio/base_events.py:184] created at /Users/runner/work/hikari/hikari/.nox/pytest/lib/python3.9/site-packages/pytest_asyncio/plugin.py:445> took 0.207 seconds
```

--- a/hikari/internal/collections.py
+++ b/hikari/internal/collections.py
@@ -30,7 +30,6 @@ __all__: typing.List[str] = [
     "SnowflakeSet",
     "ExtendedMutableMapping",
     "FreezableDict",
-    "TimedCacheMap",
     "LimitedCapacityCacheMap",
     "get_index_or_slice",
 ]
@@ -38,10 +37,8 @@ __all__: typing.List[str] = [
 import abc
 import array
 import bisect
-import datetime
 import itertools
 import sys
-import time
 import typing
 
 from hikari import snowflakes
@@ -157,92 +154,6 @@ class _FrozenDict(typing.MutableMapping[KeyT, ValueT]):
 
     def __setitem__(self, key: KeyT, value: ValueT) -> None:
         self._source[key] = (0.0, value)
-
-
-class TimedCacheMap(ExtendedMutableMapping[KeyT, ValueT]):
-    """A most-recently-inserted limited mutable mapping implementation.
-
-    This will remove entries on modification as as they pass the expiry limit.
-
-    Parameters
-    ----------
-    expiry : datetime.timedelta
-        The timedelta of how long entries should be stored for before removal.
-
-    Other Parameters
-    ----------------
-    source : typing.Optional[typing.Dict[KeyT, typing.Tuple[builtins.float, ValueT]]
-        A source dictionary of keys to tuples of float timestamps and values to
-        create this from.
-    on_expire : typing.Optional[typing.Callable[[ValueT], None]]
-        A function to call each time an item is garbage collected from this
-        map. This should take one positional argument of the same type stored
-        in this mapping as the value and should return `builtins.None`.
-
-        This will always be called after the entry has been removed.
-    """
-
-    __slots__: typing.Sequence[str] = ("_data", "_expiry", "_on_expire")
-
-    def __init__(
-        self,
-        source: typing.Optional[typing.Dict[KeyT, typing.Tuple[float, ValueT]]] = None,
-        /,
-        *,
-        expiry: datetime.timedelta,
-        on_expire: typing.Optional[typing.Callable[[ValueT], None]] = None,
-    ) -> None:
-        if expiry <= datetime.timedelta():
-            raise ValueError("expiry time must be greater than 0 microseconds.")
-
-        self._expiry: float = expiry.total_seconds()
-        self._data = source or {}
-        self._on_expire = on_expire
-        self._garbage_collect()
-
-    def clear(self) -> None:
-        self._data.clear()
-
-    def copy(self) -> TimedCacheMap[KeyT, ValueT]:
-        return TimedCacheMap(
-            self._data.copy(), expiry=datetime.timedelta(seconds=self._expiry), on_expire=self._on_expire
-        )
-
-    def freeze(self) -> typing.MutableMapping[KeyT, ValueT]:
-        return _FrozenDict(self._data.copy())
-
-    def _garbage_collect(self) -> None:
-        current_time = time.perf_counter()
-        for key, value in tuple(self._data.items()):
-            if current_time - value[0] < self._expiry:
-                break
-
-            del self._data[key]
-
-            if self._on_expire:
-                self._on_expire(value[1])
-
-    def __delitem__(self, key: KeyT) -> None:
-        del self._data[key]
-        self._garbage_collect()
-
-    def __getitem__(self, key: KeyT) -> ValueT:
-        return self._data[key][1]
-
-    def __iter__(self) -> typing.Iterator[KeyT]:
-        return iter(self._data)
-
-    def __len__(self) -> int:
-        return len(self._data)
-
-    def __setitem__(self, key: KeyT, value: ValueT) -> None:
-        #  Seeing as we rely on insertion order in _garbage_collect, we have to make sure that each item is added to
-        #  the end of the dict.
-        if key in self:
-            del self[key]
-
-        self._data[key] = (time.perf_counter(), value)
-        self._garbage_collect()
 
 
 class LimitedCapacityCacheMap(ExtendedMutableMapping[KeyT, ValueT]):

--- a/tests/hikari/hikari_test_helpers.py
+++ b/tests/hikari/hikari_test_helpers.py
@@ -150,10 +150,6 @@ def skip_on_system(os_name: str):
     return decorator
 
 
-async def idle(for_=REASONABLE_SLEEP_TIME, /):
-    await asyncio.sleep(for_)
-
-
 @contextlib.contextmanager
 def ensure_occurs_quickly():
     with async_timeout.timeout(REASONABLE_QUICK_RESPONSE_TIME):

--- a/tests/hikari/impl/test_buckets.py
+++ b/tests/hikari/impl/test_buckets.py
@@ -151,8 +151,6 @@ class TestRESTBucketManager:
         with buckets.RESTBucketManager(max_rate_limit=float("inf")) as mgr:
             assert mgr.gc_task is None
             mgr.start()
-            mgr.start()
-            mgr.start()
             assert mgr.gc_task is not None
 
     @pytest.mark.asyncio()
@@ -165,36 +163,38 @@ class TestRESTBucketManager:
             close.assert_called()
 
     @pytest.mark.asyncio()
-    async def test_gc_polls_until_closed_event_set(self):
-        # This is shit, but it is good shit.
+    async def test_gc_polls_until_closed_event_set(self, event_loop):
         with buckets.RESTBucketManager(max_rate_limit=float("inf")) as mgr:
-            mgr.start(0.01)
-            assert mgr.gc_task is not None
-            assert not mgr.gc_task.done()
-            await hikari_test_helpers.idle()
-            assert mgr.gc_task is not None
-            assert not mgr.gc_task.done()
-            await hikari_test_helpers.idle()
+            # Start the gc and initial assertions
+            task = event_loop.create_task(mgr.gc(0.001, float("inf")))
+            assert not task.done()
+
+            # [First poll] event not set => shouldn't complete the task
+            await asyncio.sleep(0.001)
+            assert not task.done()
+
+            # [Second poll] event not set during poll => shouldn't complete the task
+            await asyncio.sleep(0.001)
             mgr.closed_event.set()
-            assert mgr.gc_task is not None
-            assert not mgr.gc_task.done()
-            task = mgr.gc_task
-            await hikari_test_helpers.idle()
-            assert mgr.gc_task is None
+            assert not task.done()
+
+            # [Third poll] event set => should complete the task
+            await asyncio.sleep(0.001)
             assert task.done()
 
     @pytest.mark.asyncio()
     async def test_gc_calls_do_pass(self):
+        class ExitError(Exception):
+            ...
+
         with hikari_test_helpers.mock_class_namespace(buckets.RESTBucketManager, slots_=False)(
             max_rate_limit=float("inf")
         ) as mgr:
-            mgr.do_gc_pass = mock.Mock()
-            mgr.start(0.01, 33)
-            try:
-                await hikari_test_helpers.idle()
-                mgr.do_gc_pass.assert_called_with(33)
-            finally:
-                mgr.gc_task.cancel()
+            mgr.do_gc_pass = mock.Mock(side_effect=ExitError)
+            with pytest.raises(ExitError):
+                await mgr.gc(0.001, 33)
+
+            mgr.do_gc_pass.assert_called_with(33)
 
     @pytest.mark.asyncio()
     async def test_do_gc_pass_any_buckets_that_are_empty_but_still_rate_limited_are_kept_alive(self):

--- a/tests/hikari/impl/test_event_manager_base.py
+++ b/tests/hikari/impl/test_event_manager_base.py
@@ -167,9 +167,7 @@ class TestEventStream:
     @pytest.mark.asyncio()
     @hikari_test_helpers.timeout()
     async def test___anext___times_out(self):
-        streamer = event_manager_base.EventStream(
-            mock.Mock(), event_type=base_events.Event, timeout=hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME
-        )
+        streamer = event_manager_base.EventStream(mock.Mock(), event_type=base_events.Event, timeout=0.001)
 
         with streamer:
             with pytest.raises(LookupError):
@@ -179,18 +177,22 @@ class TestEventStream:
     @hikari_test_helpers.timeout()
     async def test___anext___waits_for_next_event(self):
         mock_event = object()
-        streamer = event_manager_base.EventStream(
-            mock.Mock(), event_type=base_events.Event, timeout=hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 3
-        )
+        streamer = event_manager_base.EventStream(mock.Mock(), event_type=base_events.Event, timeout=None)
 
-        async def add_event():
-            await asyncio.sleep(hikari_test_helpers.REASONABLE_SLEEP_TIME)
-            await streamer._listener(mock_event)
-
-        asyncio.create_task(add_event())
+        async def quickly_run_task(task):
+            try:
+                await asyncio.wait_for(asyncio.shield(task), timeout=0.01)
+            except asyncio.TimeoutError:
+                pass
 
         with streamer:
-            assert await streamer.next() is mock_event
+            next_task = asyncio.create_task(streamer.next())
+            await quickly_run_task(next_task)
+            assert not next_task.done()
+            await streamer._listener(mock_event)
+            await quickly_run_task(next_task)
+            assert next_task.done()
+            assert next_task.result() is mock_event
 
     @pytest.mark.asyncio()
     @hikari_test_helpers.timeout()
@@ -289,7 +291,7 @@ class TestEventStream:
     @pytest.mark.asyncio()
     async def test_filter(self):
         stream = hikari_test_helpers.mock_class_namespace(event_manager_base.EventStream)(
-            event_manager=mock.Mock(), event_type=base_events.Event, timeout=1
+            event_manager=mock.Mock(), event_type=base_events.Event, timeout=0.001
         )
         stream._filters = iterators.All(())
         first_pass = mock.Mock(attr=True)
@@ -312,7 +314,7 @@ class TestEventStream:
     @pytest.mark.asyncio()
     async def test_filter_handles_calls_while_active(self):
         stream = hikari_test_helpers.mock_class_namespace(event_manager_base.EventStream)(
-            event_manager=mock.Mock(), event_type=base_events.Event, timeout=1
+            event_manager=mock.Mock(), event_type=base_events.Event, timeout=0.001
         )
         stream._filters = iterators.All(())
         first_pass = mock.Mock(attr=True)

--- a/tests/hikari/impl/test_rate_limits.py
+++ b/tests/hikari/impl/test_rate_limits.py
@@ -21,11 +21,8 @@
 # SOFTWARE.
 import asyncio
 import contextlib
-import logging
 import math
-import statistics
 import sys
-import threading
 import time
 
 import mock
@@ -301,106 +298,52 @@ class TestWindowedBurstRateLimiter:
 
     @pytest.mark.asyncio()
     async def test_throttle_consumes_queue(self, event_loop):
-        with rate_limits.WindowedBurstRateLimiter(__name__, 0.01, 1) as rl:
-            rl.queue = [event_loop.create_future() for _ in range(15)]
-            old_queue = list(rl.queue)
-            await rl.throttle()
+        with mock.patch.object(asyncio, "sleep"):
+            with rate_limits.WindowedBurstRateLimiter(__name__, 0.001, 1) as rl:
+                rl.queue = [event_loop.create_future() for _ in range(15)]
+                old_queue = list(rl.queue)
+                await rl.throttle()
 
         assert len(rl.queue) == 0
         for i, future in enumerate(old_queue):
             assert future.done(), f"future {i} was incomplete!"
 
     @pytest.mark.asyncio()
-    @hikari_test_helpers.timeout(20)
-    @hikari_test_helpers.retry(5)
     async def test_throttle_when_limited_sleeps_then_bursts_repeatedly(self, event_loop):
-        # Schedule concurrently but do not break our timeout.
-        # We should retry a few times, as CI runners may run too slowly if
-        # under load. As much as we try, this will always be time-relative,
-        # which is not a good test, but too much mocking is needed to obfuscate
-        # out that detail.
-        # TODO: find a better way of doing this?
-        await event_loop.run_in_executor(None, self._run_test_throttle_logic)
+        window = 5
+        sleep_count = 0
+        futures = [event_loop.create_future() for _ in range(20)]
 
-    def _run_test_throttle_logic(self):
-        threads = [
-            threading.Thread(
-                target=self._run_test_throttle_logic_on_this_thread,
-            )
-            for _ in range(20)
-        ]
+        async def mock_sleep(t):
+            nonlocal sleep_count
 
-        for thread in threads:
-            thread.start()
+            for i, future in enumerate(futures):
+                if i >= (window * sleep_count):
+                    assert not future.done(), f"future {i} was complete, expected it to be incomplete!"
+                else:
+                    assert future.done(), f"future {i} was incomplete, expected it to be completed!"
 
-        for thread in threads:
-            thread.join()
+            sleep_count += 1
 
-    def _run_test_throttle_logic_on_this_thread(self):
-        event_loop = asyncio.new_event_loop()
-        asyncio.set_event_loop(event_loop)
-        try:
-            event_loop.run_until_complete(self._run_test_throttle_logic_on_loop(event_loop))
-        finally:
-            event_loop.close()
+        stack = contextlib.ExitStack()
+        rl = stack.enter_context(rate_limits.WindowedBurstRateLimiter(__name__, 0, window))
+        stack.enter_context(mock.patch.object(asyncio, "sleep", new=mock_sleep))
 
-    @staticmethod
-    async def _run_test_throttle_logic_on_loop(event_loop):
-        limit = 2
-        period = 1.5
-        total_requests = int(period * limit * 2)
-        max_distance_within_window = 0.05
-        completion_times = []
-        logger = logging.getLogger(__name__)
-
-        def create_task(i):
-            logger.info("making task %s", i)
-            future = event_loop.create_future()
-            future.add_done_callback(lambda _: completion_times.append(time.perf_counter()))
-            return future
-
-        with rate_limits.WindowedBurstRateLimiter(__name__, period, limit) as rl:
-            futures = [create_task(i) for i in range(total_requests)]
+        with stack:
             rl.queue = list(futures)
             rl.reset_at = time.perf_counter()
-            logger.info("throttling back")
             await rl.throttle()
             # die if we take too long...
-            logger.info("waiting for stuff to finish")
-            await asyncio.wait(futures, timeout=period * limit + period)
+            await asyncio.wait(futures, timeout=3)
 
-        assert (
-            len(completion_times) == total_requests
-        ), f"expected {total_requests} completions but got {len(completion_times)}"
-
-        # E203 - Whitespace before ":". Black reformats it
-        windows = [completion_times[i : i + limit] for i in range(0, total_requests, limit)]  # noqa: E203
-
-        for i, window in enumerate(windows):
-            logger.info("window %s %s", i, window)
-            mode = statistics.mode(window)
-            for j, element in enumerate(window):
-                assert math.isclose(element, mode, abs_tol=max_distance_within_window), (
-                    f"not close! windows[{i}][{j}], future {i * len(window) + j}, "
-                    f"val {element}, mode {mode}, max diff {max_distance_within_window}"
-                )
-
-        assert len(windows) >= 3, "not enough windows to sample correctly"
-        assert len(windows[0]) > 1, "not enough datapoints per window to sample correctly"
-
-        for i in range(1, len(windows)):
-            previous_last = windows[i - 1][-1]
-            next_first = windows[i][0]
-            logger.info("intra-window index=%s value=%s versus index=%s value=%s", i - 1, previous_last, i, next_first)
-
-            assert math.isclose(next_first - previous_last, period, abs_tol=max_distance_within_window), (
-                f"distance between windows is not acceptable! {i - 1}={previous_last} {i}={next_first}, "
-                f"max diff = {max_distance_within_window}"
-            )
+        assert sleep_count == 4
+        assert len(rl.queue) == 0
+        for i, future in enumerate(futures):
+            assert future.done(), f"future {i} was incomplete!"
 
     @pytest.mark.asyncio()
     async def test_throttle_resets_throttle_task(self, event_loop):
-        with rate_limits.WindowedBurstRateLimiter(__name__, 0.01, 1) as rl:
+        with rate_limits.WindowedBurstRateLimiter(__name__, 0.001, 1) as rl:
             rl.queue = [event_loop.create_future() for _ in range(15)]
             rl.throttle_task = None
             await rl.throttle()

--- a/tests/hikari/impl/test_rest.py
+++ b/tests/hikari/impl/test_rest.py
@@ -213,21 +213,26 @@ class TestClientCredentialsStrategy:
         assert new_token == "Bearer okokok.fofofo.ddd"  # noqa S105: Possible Hardcoded password
 
     async def test_acquire_uses_newly_cached_token_after_acquiring_lock(self):
-        token = "abc.abc.abc"  # noqa S105: Possible Hardcoded password
+        class MockLock:
+            def __init__(self, strategy):
+                self._strategy = strategy
+
+            async def __aenter__(self):
+                self._strategy._token = "abc.abc.abc"
+                return self
+
+            async def __aexit__(self, exc_type, exc_val, exc_tb):
+                return
+
         mock_rest = mock.AsyncMock()
         strategy = rest.ClientCredentialsStrategy(client=65123, client_secret="12354")
+        strategy._lock = MockLock(strategy)
+        strategy._token = None
+        strategy._expire_at = time.monotonic() + 500
 
-        async def hold_strategy():
-            async with strategy._lock:
-                await asyncio.sleep(hikari_test_helpers.REASONABLE_SLEEP_TIME)
-                strategy._token = token
-                strategy._expire_at = time.monotonic() + 600
-
-        asyncio.create_task(hold_strategy())
-        await asyncio.sleep(hikari_test_helpers.REASONABLE_SLEEP_TIME // 1000)
         result = await strategy.acquire(mock_rest)
 
-        assert result == token
+        assert result == "abc.abc.abc"
 
         mock_rest.authorize_client_credentials_token.assert_not_called()
 

--- a/tests/hikari/internal/test_aio.py
+++ b/tests/hikari/internal/test_aio.py
@@ -39,7 +39,8 @@ class CoroutineStub:
 
     def __await__(self):
         self.awaited = True
-        return hikari_test_helpers.idle().__await__()
+        # asyncio.sleep(0) is a special case that will just quickly return without doing anything special
+        return asyncio.sleep(0)
 
     def __repr__(self):
         args = ", ".join(map(repr, self.args))
@@ -311,23 +312,29 @@ class TestAllOf:
         f2 = event_loop.create_future()
         f3 = event_loop.create_future()
 
+        async def quickly_run_task(task):
+            try:
+                await asyncio.wait_for(asyncio.shield(task), timeout=0.01)
+            except asyncio.TimeoutError:
+                pass
+
         waiter = event_loop.create_task(aio.all_of(f1, f2, f3))
-        await hikari_test_helpers.idle(0.1)
+        await quickly_run_task(waiter)
         assert not waiter.cancelled()
         assert not waiter.done()
 
         f1.set_result(1)
-        await hikari_test_helpers.idle(0.1)
+        await quickly_run_task(waiter)
         assert not waiter.cancelled()
         assert not waiter.done()
 
         f2.set_result(2)
-        await hikari_test_helpers.idle(0.1)
+        await quickly_run_task(waiter)
         assert not waiter.cancelled()
         assert not waiter.done()
 
         f3.set_result(3)
-        await hikari_test_helpers.idle(0.1)
+        await quickly_run_task(waiter)
         assert not waiter.cancelled()
         assert waiter.done()
 
@@ -340,8 +347,6 @@ class TestAllOf:
         f3 = event_loop.create_future()
 
         waiter = event_loop.create_task(aio.all_of(f1, f2, f3))
-        # Wait a little bit and ensure asyncio doesn't kill it for some reason
-        await asyncio.sleep(0.25)
         assert not waiter.cancelled()
         assert not waiter.done()
 
@@ -360,7 +365,7 @@ class TestAllOf:
         f2 = event_loop.create_future()
         f3 = event_loop.create_future()
 
-        waiter = event_loop.create_task(aio.all_of(f1, f2, f3, timeout=0.1))
+        waiter = event_loop.create_task(aio.all_of(f1, f2, f3, timeout=0.01))
         with pytest.raises(asyncio.TimeoutError):
             await waiter
         assert not waiter.cancelled(), "future was forcefully cancelled?"
@@ -375,9 +380,10 @@ class TestAllOf:
         f3 = event_loop.create_future()
 
         waiter = event_loop.create_task(aio.all_of(f1, f2, f3))
-        event_loop.call_later(0.01, lambda *_: waiter.cancel())
+        event_loop.call_later(0.001, lambda *_: waiter.cancel())
         with pytest.raises(asyncio.CancelledError):
             await waiter
+
         assert waiter.cancelled(), "future was forcefully cancelled?"
         assert waiter.done(), "asyncio.CancelledError not raised?"
 

--- a/tests/hikari/internal/test_aio.py
+++ b/tests/hikari/internal/test_aio.py
@@ -365,7 +365,7 @@ class TestAllOf:
         f2 = event_loop.create_future()
         f3 = event_loop.create_future()
 
-        waiter = event_loop.create_task(aio.all_of(f1, f2, f3, timeout=0.01))
+        waiter = event_loop.create_task(aio.all_of(f1, f2, f3, timeout=0.001))
         with pytest.raises(asyncio.TimeoutError):
             await waiter
         assert not waiter.cancelled(), "future was forcefully cancelled?"

--- a/tests/hikari/internal/test_collections.py
+++ b/tests/hikari/internal/test_collections.py
@@ -20,17 +20,13 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 import array as array_
-import asyncio
-import datetime
 import operator
 import sys
-import time
 
 import mock
 import pytest
 
 from hikari.internal import collections
-from tests.hikari import hikari_test_helpers
 
 
 class TestFreezableDict:
@@ -111,147 +107,6 @@ class TestFrozenDict:
         mock_map["foo bar"] = 42
 
         assert mock_map == {"rororo": "bye 3231", "2121": "4321", "foo bar": 42}
-
-
-class TestTimedCacheMap:
-    def test___init___with_source(self):
-        raw_map = {
-            "not_in": (time.perf_counter() - 50, "goodbye"),
-            "ok": (time.perf_counter() + 30, "no"),
-            "blam": (time.perf_counter() + 20, "bye"),
-        }
-        mock_map = collections.TimedCacheMap(raw_map, expiry=datetime.timedelta(seconds=42))
-
-        assert mock_map == {"blam": "bye", "ok": "no"}
-
-    def test___init___raises_value_error_on_invalid_expiry(self):
-        with pytest.raises(ValueError, match="expiry time must be greater than 0 microseconds."):
-            collections.TimedCacheMap(expiry=datetime.timedelta(seconds=0))
-
-        with pytest.raises(ValueError, match="expiry time must be greater than 0 microseconds."):
-            collections.TimedCacheMap(expiry=datetime.timedelta(seconds=-50))
-
-    def test_clear(self):
-        raw_map = {
-            "floom": (999999999999999999999999, "buebue"),
-            "bash": (999999999999999999999999, "bunny_time"),
-        }
-        mock_map = collections.TimedCacheMap(raw_map, expiry=datetime.timedelta(seconds=4523412))
-        mock_map.clear()
-
-        assert mock_map._data == {}
-
-    def test_copy(self):
-        raw_map = {
-            "floom": (999999999999999999999999, "buebue"),
-            "bash": (999999999999999999999999, "bunny_time"),
-        }
-        mock_map = collections.TimedCacheMap(raw_map, expiry=datetime.timedelta(seconds=4523412))
-        result = mock_map.copy()
-
-        assert result is not mock_map
-        assert isinstance(result, collections.TimedCacheMap)
-
-        assert result == {"floom": "buebue", "bash": "bunny_time"}
-
-    def test_freeze(self):
-        raw_map = {
-            "bash": (999999999999999999999999, "gtuutueu"),
-            "blam": (999999999999999999999999, "poke"),
-            "owowo": (999999999999999999999999, "no you"),
-        }
-        mock_map = collections.TimedCacheMap(raw_map, expiry=datetime.timedelta(seconds=6523423))
-        result = mock_map.freeze()
-
-        assert result == {"bash": "gtuutueu", "blam": "poke", "owowo": "no you"}
-        assert isinstance(result, collections._FrozenDict)
-
-    def test___delitem__(self):
-        mock_map = collections.TimedCacheMap(expiry=datetime.timedelta(seconds=100))
-        mock_map.update({"ok": "no", "ayanami": "rei qt"})
-        del mock_map["ok"]
-        assert mock_map == {"ayanami": "rei qt"}
-
-    @pytest.mark.asyncio()
-    async def test___delitem___garbage_collection(self):
-        mock_map = collections.TimedCacheMap(
-            expiry=datetime.timedelta(seconds=hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 3)
-        )
-        mock_map.update({"nyaa": "see", "awwo": "awoo2"})
-        await asyncio.sleep(hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 2)
-        assert mock_map == {"nyaa": "see", "awwo": "awoo2"}
-        mock_map.update({"ayanami": "shinji", "rei": "aww"})
-        assert mock_map == {"nyaa": "see", "awwo": "awoo2", "ayanami": "shinji", "rei": "aww"}
-        await asyncio.sleep(hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 2)
-        del mock_map["ayanami"]
-        assert mock_map == {"rei": "aww"}
-
-    @pytest.mark.asyncio()
-    async def test___delitem___garbage_collection_on_expire_set(self):
-        expire_call = mock.Mock()
-        mock_map = collections.TimedCacheMap(
-            expiry=datetime.timedelta(seconds=hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 3),
-            on_expire=expire_call,
-        )
-        mock_map.update({"nyaa": "see", "awwo": "awoo2"})
-        await asyncio.sleep(hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 6)
-        expire_call.assert_not_called()
-        mock_map.update({"ayanami": "shinji", "rei": "aww"})
-        expire_call.assert_has_calls((mock.call("see"), mock.call("awoo2")))
-
-    def test___getitem___for_valid_entry(self):
-        mock_map = collections.TimedCacheMap(expiry=datetime.timedelta(seconds=100))
-        mock_map["OK"] = 42
-        mock_map["blam"] = 8
-        assert mock_map["OK"] == 42
-
-    def test___getitem___for_unknown_entry(self):
-        mock_map = collections.TimedCacheMap(expiry=datetime.timedelta(seconds=100))
-        mock_map["blam"] = 8
-
-        with pytest.raises(KeyError):
-            mock_map["OK"]
-
-    def test___iter__(self):
-        mock_map = collections.TimedCacheMap(expiry=datetime.timedelta(seconds=100))
-        mock_map.update({"o": "k", "k": "o", "awoo": "blam", "hikari": "rei"})
-        assert list(mock_map) == ["o", "k", "awoo", "hikari"]
-
-    def test___len__(self):
-        mock_map = collections.TimedCacheMap(expiry=datetime.timedelta(seconds=100))
-        mock_map.update({"o": "k", "boop": "bop", "k": "o", "awoo": "blam", "rei": "cute", "hikari": "rei"})
-        assert len(mock_map) == 6
-
-    def test___setitem__(self):
-        mock_map = collections.TimedCacheMap(expiry=datetime.timedelta(seconds=100))
-        mock_map["blat"] = 42
-        assert mock_map == {"blat": 42}
-
-    def test___setitem___removes_old_entry_instead_of_replacing(self):
-        mock_map = collections.TimedCacheMap(
-            {
-                "ok": (time.perf_counter() + 50, "no"),
-                "bar": (time.perf_counter() + 60, "bat"),
-                "foo": (time.perf_counter() + 70, "blam"),
-            },
-            expiry=datetime.timedelta(seconds=100),
-        )
-        mock_map["ok"] = "foo"
-        assert list(mock_map.items())[2] == ("ok", "foo")
-
-    @pytest.mark.asyncio()
-    async def test___setitem___garbage_collection(self):
-        mock_map = collections.TimedCacheMap(
-            expiry=datetime.timedelta(seconds=hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 3)
-        )
-        mock_map["OK"] = "no"
-        await asyncio.sleep(hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 2)
-        assert mock_map == {"OK": "no"}
-        mock_map["ayanami"] = "rei"
-        assert mock_map == {"OK": "no", "ayanami": "rei"}
-        await asyncio.sleep(hikari_test_helpers.REASONABLE_QUICK_RESPONSE_TIME * 2)
-        mock_map["nyaa"] = "qt"
-        assert mock_map == {"ayanami": "rei", "nyaa": "qt"}
 
 
 class TestLimitedCapacityCacheMap:

